messagebus/src/vespa/messagebus/network/rpcsend.cpp

```diff
@@ -0,0 +1,281 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "rpcsend.h"
+#include "rpcsend_private.h"
+#include "rpcserviceaddress.h"
+#include <vespa/messagebus/network/rpcnetwork.h>
+#include <vespa/messagebus/tracelevel.h>
+#include <vespa/messagebus/emptyreply.h>
+#include <vespa/messagebus/errorcode.h>
+#include <vespa/vespalib/util/stringfmt.h>
+#include <vespa/fnet/channel.h>
+#include <vespa/fnet/frt/reflection.h>
+
+#include <vespa/vespalib/data/slime/cursor.h>
+
+using vespalib::make_string;
+
+namespace mbus {
+
+using network::internal::ReplyContext;
+using network::internal::SendContext;
+
+namespace {
+
+class FillByCopy final : public PayLoadFiller
+{
+public:
+    FillByCopy(BlobRef payload) : _payload(payload) { }
+    void fill(FRT_Values & v) const override {
+        v.AddData(_payload.data(), _payload.size());
+    }
+    void fill(const vespalib::Memory & name, vespalib::slime::Cursor & v) const override {
+        v.setData(name, vespalib::Memory(_payload.data(), _payload.size()));
+    }
+private:
+    BlobRef _payload;
+};
+
+class FillByHandover final : public PayLoadFiller
+{
+public:
+    FillByHandover(Blob payload) : _payload(std::move(payload)) { }
+    void fill(FRT_Values & v) const override {
+        v.AddData(std::move(_payload.payload()), _payload.size());
+    }
+    void fill(const vespalib::Memory & name, vespalib::slime::Cursor & v) const override {
+        v.setData(name, vespalib::Memory(_payload.data(), _payload.size()));
```

Comment:
I expect the new external memory slime data value will be used here?

Reply:
Correct, here we add the serialized blob from the protocol (documentapi) as an opaque blob.